slu/slu/utils/config.py

```diff
@@ -172,6 +174,132 @@ def generate(self) -> Dict[str, Config]:
+        supported_languages: list = list(config_dict.keys())
+        return supported_languages
+
+    def _get_config_path(self) -> str:
+        return self.config_path
+
+    def _valid_string(self, string: str) -> bool:
+        if isinstance(string, str):
+            if all(
+                [
+                    len(string) > 0,
+                    string != "nan",
+                    string != ".nan",
+                    string != "",
+                    string != " ",
+                    "Unnamed" not in string,
+                ]
+            ):
+                return True
+        return False
+
+    def validate(self) -> None:
+        if not (all(isinstance(lang, str) for lang in self.config_dict.keys())):
+            raise TypeError(
+                f"Invalid format, please make sure prompts.yaml is correctly defined"
+            )
+
+        for lang in self.config_dict:
+            if not (
+                all(isinstance(nls_label, str) for nls_label in self.config_dict[lang])
+                & all(
+                    self._valid_string(nls_label)
+                    for nls_label in self.config_dict[lang]
+                )
+            ):
+                raise TypeError(
+                    f"Invalid or Malformed nls_label name, please make sure prompts.yaml is correctly defined"
+                )
+
+            for nls_label in self.config_dict[lang]:
+                if not self._valid_string(self.config_dict[lang][nls_label]):
+                    raise TypeError(
+                        f"Invalid or Malformed prompt encountered for nls_label: {nls_label} in lang: {lang}"
+                    )
+
+    def get_prompt(
+        self, lang: str, nls_label: str, return_all: bool = False
+    ) -> List[str]:
+        if not lang in self.supported_languages:
+            error_message = (
+                f"No prompts found for language {lang}, please check the config."
+            )
+            raise KeyError(error_message)
+
+        if not nls_label in self.config_dict[lang]:
+            self.missing_nls_labels.add(nls_label)
+            if self.debug:
+                logger.debug(
+                    f"NLS Label {nls_label} not found in config for language {lang}"
+                )
+            return [self.null_prompt_token]
+
+        if not self._valid_string(self.config_dict[lang][nls_label]):
+            if self.debug:
+                logger.debug(f"No prompt found for NLS Label {nls_label}")
+            return [self.null_prompt_token]
+
+        return (
+            self.config_dict[lang][nls_label]
+            if return_all
+            else random.sample(self.config_dict[lang][nls_label], 1)
+        )
+
+    def lookup_prompt(
+        self, lang: str, nls_label: str, return_all: bool = False
+    ) -> List[str]:
+        """
+        Same as get_prompt() method, but built for faster lookup to reduce latency during inference.
```

Comment:
I see that it skips a lot of operations that `get_prompt` has. What is the effect of that?